### PR TITLE
Update ADLSGen2-HOWTO.md

### PR DESCRIPTION
**Pull request by dgcaron**
# Description
I had a hard time figuring out how to connect to a delta table that is stored in ADLS Gen2 and only found a way by digging into the source code. I would like to save other people the same trouble by adding this to the docs.

# Related Issue(s)

# Documentation

### CODE REVIEW

File: docs/ADLSGen2-HOWTO.md
Comment (wjones127) [inline docs/ADLSGen2-HOWTO.md:47]: Thanks! Would you like to also add this to the [usage docs](https://delta-io.github.io/delta-rs/python/usage.html#loading-a-delta-table) as well?

It looks like I might have gotten the wrong prefix when I add this :facepalm::

https://github.com/delta-io/delta-rs/blame/main/python/docs/source/usage.rst#L59-L60

**Review by thovoll — DISMISSED** [review]

File: docs/ADLSGen2-HOWTO.md
Comment (zeevm) [reply docs/ADLSGen2-HOWTO.md:47]: Why use this non-standard uri schema for ADLS?
Should either use abfss if the underlying storage layer "knows" adls or use https with the appropriate blob storage endpoint that underlies every adls account.

Comment (wjones127) [reply docs/ADLSGen2-HOWTO.md:47]: Actually, googling a bit, it seems like `abfss` is somewhat standard: https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-introduction-abfs-uri
Thanks @zeevm 

@dgcaron Does that schema work for you?

Comment (thovoll) [reply docs/ADLSGen2-HOWTO.md:47]: `abfs[s]` refers to the Hadoop File System driver (see the docs linked by @wjones127), which is not being used here.

Using `https` would be inconsistent with the fact that delta-rs accepts a URI with a cloud specific scheme ("s3://" and "gs://") for the Amazon and Google backends.

Comment (thovoll) [reply docs/ADLSGen2-HOWTO.md:47]: > Why use this non-standard uri schema for ADLS? Should either use abfss if the underlying storage layer "knows" adls or use https with the appropriate blob storage endpoint that underlies every adls account.

delta-rs relies on the atomic rename functionality of ADLS Gen2 and cannot work with "plain" Blob Storage.

**Comment by dgcaron** [comment]
Abfss was what I tried initially, unsuspecting users that will not read the manual will most likely do the same.  After failing and not finding the right scheme I went into the code to see what it expects. To save anyone else the search I thought it was a good thing to add it to the docs.

**Review by houqp — APPROVED** [review]

**Comment by houqp** [comment]
@dgcaron do you want to include the python usage docs update in this PR as well?

**Comment by dgcaron** [comment]
Yes sure, I'll do that today

**Comment by zeevm** [comment]
I'd think fixing the docs isn't the issue, the implementation should be fixed to use either abfss if delta-rs knows how to talk to the adls endpoint or https if it talks to the blob storage endpoint, 'adls2' is never the right scheme to use

**Comment by thovoll** [comment]
> I'd think fixing the docs isn't the issue, the implementation should be fixed to use either abfss if delta-rs knows how to talk to the adls endpoint or https if it talks to the blob storage endpoint, 'adls2' is never the right scheme to use

Hi @zeevm, I addressed all these points above, stated again here with some more elaboration:

1. `abfs[s]://` refers to the Hadoop File System driver, which is not being used by delta-rs. From [the Azure docs](https://docs.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-introduction-abfs-uri): "The Hadoop Filesystem driver that is compatible with Azure Data Lake Storage Gen2 _is known by its scheme identifier abfs_ (Azure Blob File System). Consistent with other Hadoop Filesystem drivers, the ABFS driver employs a URI format to address files and directories within a Data Lake Storage Gen2 capable account."

1. Using `https` URLs would be inconsistent with the fact that delta-rs accepts URIs with a cloud specific scheme (`s3://` and `gs://`) for the Amazon and Google backends. 

Based on those two points, it makes sense to use a cloud specific URI scheme in delta-rs, but not `abfs[s]://` - hence: `adls2://`. 

Also: the delta-rs Azure backend relies on the atomic rename functionality of Azure Data Lake Storage (ADLS) Gen2 and doesn't work with Azure Blob Storage.

The `adls2://` URI scheme is an invention of delta-rs, but there are other libraries that have done something similar for Azure storage. The root cause of this need is that ADLS Gen2 doesn't define a URI scheme other than in the context of Hadoop. It's worth noting that the [S3 Hadoop File System driver](https://hadoop.apache.org/docs/stable/hadoop-aws/tools/hadoop-aws/index.html#Introducing_the_Hadoop_S3A_client.) doesn't use `s3://` but `s3a://` - the reasons are complicated but this supports introducing `adls2://` to distinguish from `abfs[s]://`.

**Review by wjones127 — APPROVED** [review]

**Comment by dgcaron** [comment]
did i break the build here?

**Comment by thovoll** [comment]
> did i break the build here?

Clippy errors can show up because of toolchain updates. Take a look at the error, it has a suggestion to try and fix the clippy error.

**Comment by dgcaron** [comment]
yeah, i'll wait for your merge as it is a linter error for files I didn't touch at all.

**Comment by thovoll** [comment]
> yeah, i'll wait for your merge as it is a linter error for files I didn't touch at all.

Actually, @houqp already fixed these yesterday. Just fetch upstream on your branch.

**Comment by dgcaron** [comment]
do i need to do something additional to get this merged?

**Comment by thovoll** [comment]
> do i need to do something additional to get this merged?

No, someone with merge privileges has to review, approve, and merge.

**Comment by houqp** [comment]
Sorry, there were some intermittent CI errors, I was waiting for the rerun of the CI before doing a merge. Thanks again @dgcaron for the fix! And than kyou @thovoll and @wjones127 for the review :)

**Pull request merged**
